app.rb

```diff
@@ -0,0 +1,166 @@
+      puts '[i] No one here!'
+    else
+      puts '# List of All People'
+      list_all(@people)
+    end
+  end
+
+  def create_person
+    puts '# Creating New Person'
+    role = get_input("Select person's role ::[1]Student; [2]Teacher::", options: [1, 2])
+    role = role == 1 ? 'student' : 'teacher'
+    name = get_input('Enter Name')
+    age = get_input('Enter age', type: 'num')
+    case role
+    when 'student'
+      create_student(name, age)
+    when 'teacher'
+      create_teacher(name, age)
+    end
+  end
+
+  def create_book
+    puts '# Creating New Book'
+    title = get_input('Enter title')
+    author = get_input('Enter author')
+    book = Book.new(title, author)
+    @books.push(book)
+    puts '[k] New book created successfully!', book.to_s
+  end
+
+  def create_rental
+    if @books.empty?
+      puts '[i] Add books to make a rental'
+      return
+    elsif @people.empty?
+      puts '[i] Add people to make a rental'
+      return
+    end
+
+    puts '# Creating New Rental'
+    list_all(@books)
+    book_index = get_input('Select a book by number', options: (1..@books.length).to_a) - 1
+    list_all(@people)
+    people_index = get_input('Select a person by number', options: (1..@people.length).to_a) - 1
+    date = get_input('Enter date', type: 'date')
+    rental = Rental.new(date, @books[book_index], @people[people_index])
+    puts '[k] Rental created successfully!'
+    puts "#{rental.date} Book: #{rental.book.title}, Borrower: #{rental.person.name}"
+  end
+
+  def list_rental_for_person
+    if @people.empty?
+      puts '[i] No one here!'
+      return
+    end
+
+    person_id = get_input("Enter person's ID", type: 'num')
+    person = @people.select { |p| p.id == person_id }[0]
+    if person.nil?
+      puts "[!] Person with ID:#{person_id} not found"
+      return
+    elsif person.rentals.empty?
+      puts "[i] #{person.name} has no rentals"
+    end
+    puts "# List of Rentals for #{person.name}"
+    person.rentals.each do |rental|
+      puts "#{rental.date} Book: #{rental.book.title} by #{rental.book.author}"
+    end
+  end
+
+  def get_input(prompt, options: [], min_length: 1, type: 'str')
+    valid = false
+    input = nil
+    until valid
+      print prompt, ' >> '
+      input = gets.chomp.strip
+      next unless valid?(input, min_length, type)
+
+      unless type == 'date'
+        input = input.to_i.zero? ? input : input.to_i
+      end
+      if !options.empty? && !options.include?(input)
+        puts '[!] Invalid Input'
+        next
+      end
+
+      valid = true
+    end
+    input
+  end
+
+  private
+
+  def create_student(name, age)
+    choice = get_input('Choose classroom ::[M]ath; [C]omputer; [H]istory::', options: %w[m c h])
+    classroom = case choice
+                when 'm'
+                  @classrooms[:math]
+                when 'c'
+                  @classrooms[:computer]
+                when 'h'
+                  @classrooms[:history]
+                end
```

Comment:
- **[OPTIONAL]** When I see the options to select a classroom I selected the capital letter but it returned an error until I selected the lowercase. Please 🙏🏽 make available to choice "M" or "m" for a better UX.

![image](https://user-images.githubusercontent.com/58642949/225615865-f8c7c240-aec6-4077-b63d-718f13ebdc31.png)